integration_tests/cairo_zero_hint_tests/find_element.small.cairo

```diff
@@ -0,0 +1,70 @@
+%builtins range_check
+from starkware.cairo.common.find_element import find_element
+from starkware.cairo.common.alloc import alloc
+
+struct MyStruct {
+    a: felt,
+    b: felt,
+}
+
+func find_element_with_generic_key{range_check_ptr}() -> () {
+    alloc_locals;
+    let (local array_ptr: felt*) = alloc();
+    assert array_ptr[0] = 1;
+
+    let (element_ptr: felt*) = find_element(
+        array_ptr=array_ptr, elm_size=1, n_elms=1, key=1
+    );
+    assert [element_ptr] = 1;
+
+    return ();
+}
+
+struct TestStruct {
+    a: felt,
+    b: felt,
+    c: felt,
+}
+
+func find_element_with_struct_key{range_check_ptr}() -> () {
+    alloc_locals;
+    let (local array_ptr: TestStruct*) = alloc();
+    assert array_ptr[0] = TestStruct(a=111, b=112, c=113);
+    assert array_ptr[1] = TestStruct(a=211, b=212, c=213);
+    assert array_ptr[2] = TestStruct(a=311, b=312, c=313);
```

Comment:
I don't think its possible since cairo memory is immutable and the only way to compare is assert, but can we perhaps modify this to test the following also ? 

`Note however that if the array has multiple elements with said key the function may return any of those elements.`

Reply:
yeah i dont see any reliable way to test this. It has not been tested for any other vm

Reply:
Right. Then I think the current tests are exhaustive.